kubernetes/linux/main.sh

```diff
@@ -334,32 +352,39 @@ if [ -e "/etc/ama-logs-secret/WSID" ]; then
       fi
 
       if [ $? -ne 0 ]; then
+            registry="https://mcr.microsoft.com/v2/"
+            if [ $CLOUD_ENVIRONMENT == "usnat" ] || [ $CLOUD_ENVIRONMENT == "ussec" ] || [ $CLOUD_ENVIRONMENT == "azurechinacloud" ]; then
+                  if [ ! -z "$MCR_URL" ]; then
```

Comment:
q1. what will will be the curl response if the MCR_URL is not set?  we should also the log message here if the MCR_URL is empty/not-set.

Reply:
Yes makes sense. Added the logic for trying curl only if registry is present. Also, added the log messaged in failure cases